web/src/pages.rs

```diff
@@ -90,35 +90,50 @@ pub async fn article_detail(
         if let Some(markdown) = markdown {
             let categories = sqlx::query_as!(
                 Category,
-                "SELECT * FROM category WHERE markdown_id = $1",
+                "SELECT categories.*
+                FROM categories
+                JOIN markdown_categories USING (category_id)
+                WHERE markdown_id = $1",
                 markdown.markdown_id
             )
             .fetch_all(&state.db)
             .await?;
-            Some((markdown, categories))
+            Some((page_snapshot, markdown, categories))
         } else {
             None
         }
     } else {
         None
     };
-    let rendered_html = if let Some((markdown, categories)) = markdown {
+    let inner_html = if let Some((page_snapshot, markdown, categories)) = stuff {
         maud::html! {
+            p { b { "Fetched At: " } (page_snapshot.fetched_at) }
             ul {
                 @for category in categories {
-                    li { b { "Category: " } (category.category.as_deref().unwrap_or("No category")) }
+                    li { b { "Category: " } (category.category) }
                 }
             }
-            h1 { b { "Title: " } (markdown.title.as_deref().unwrap_or("")) }
-            p { b { "Summary: " } (markdown.summary) }
+            @if let Some(title) = markdown.title {
+                h1 { b { "Title: " } (title) }
+            }
+            @if let Some(summary) = markdown.summary {
+                p { b { "Summary: " } (summary) }
+            } @else {
+                p data-controller="loader" { "Generating summary....." }
+            }
         }
     } else {
         maud::html! {
-            p { "Generating snapshot....." }
+            p data-controller="loader" { "Generating snapshot....." }
         }
     };
 
-    Ok(t.render(rendered_html).into_response())
+    Ok(t.render(maud::html! {
+        h1 { "Article Detail" }
+        h2 { "URL: " (article.url) }
+        div { (inner_html) }
+    })
+    .into_response())
```

Comment:
I don't really like the structure how this method...

Just the nested `if lets` and stuff makes it hard to follow

Just saying that I want to come back to how I structure these methods as I made more of them